test/e2e/helpers.go

```diff
@@ -0,0 +1,125 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"testing"
+
+	"github.com/openshift-knative/serverless-operator/test"
+	v1 "github.com/openshift/api/route/v1"
+	ioprometheusclient "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	pkgTest "knative.dev/pkg/test"
+)
+
+func extracMetrictData(body io.Reader, metricName string) (*float64, error) {
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(body)
+	if err != nil {
+		return nil, fmt.Errorf("reading text format failed: %v", err)
+	}
+	pm := prometheusMetric(metricFamilies, "knative_up")
+	if pm == nil {
+		return nil, errors.New("could not get metric family `knative_up` from prometheus exported metrics")
+	}
+	status := getMetricValueByTypeLabel(metricName, pm)
+	if status == nil {
+		return nil, fmt.Errorf("could not get metric type `%s` from prometheus metric `knative_up`", metricName)
+	}
+	return status, nil
+}
+
+func prometheusMetric(metricFamilies map[string]*ioprometheusclient.MetricFamily, key string) []*ioprometheusclient.Metric {
+	if metric, ok := metricFamilies[key]; ok && len(metric.Metric) > 0 {
+		return metric.Metric
+	}
+	return nil
+}
+
+func getMetricValueByTypeLabel(label string, metrics []*ioprometheusclient.Metric) *float64 {
+	for _, metric := range metrics {
+		if len(metric.Label) == 0 {
+			return nil
```

Comment:
Are all these `return nil` supposed to be `break` to actually search through the list?

Reply:
Yeah although there is one label will update.